### PR TITLE
fix: hide filter if no callback is provided

### LINKED ISSUE #114: VIS: don't show filter UI if no callback is given
## Short description

When the Vis component does not get any filter callback passed, the filter UI (see screenshot) in the sidebar should not be visible as it does nothing in this case. Please hide it.

image

### PR DESCRIPTION
**Pull request by dv-usama-ansari**
Closes #114 

### Developer Checklist (Definition of Done)

**Issue**

- [x] All acceptance criteria from the issue are met
- [x] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [x] Requires UI/UX/Vis review
  - [x] Reviewer(s) are notified @dvdanielamoitzi 
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [x] Branch is up-to-date with the branch to be merged with, i.e., develop
- [x] Code is cleaned up and formatted
- [ ] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [x] Add type label (e.g., *bug*, *feature*) to this pull request
- [x] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes

- The filter buttons are not visible anymore for the components which do not have a filter callback

### Screenshots

image

### Additional notes for the reviewer(s)

-  N/A
Thanks for creating this pull request 🤗

### CODE REVIEW

**Comment by dvmartinweigl** [comment]
Thx for your changes @dv-usama-ansari 
I tested your branch in Bioinsight and it seems like you currently only hide the filter buttons for the Barchart.
Please also hide the filter for all  other Vis components as well. Maybe there is a generic way to do so.

File: src/vis/bar/BarVisSidebar.tsx [hunk 1]
```diff
@@ -114,7 +96,13 @@ export function BarVisSidebar({
             <BarDirectionButtons callback={(direction: EBarDirection) => setConfig({ ...config, direction })} currentSelected={config.direction} />
           )
         : null}
-      {mergedOptionsConfig.filter.enable ? mergedOptionsConfig.filter.customComponent || <FilterButtons callback={filterCallback} /> : null}
+      {mergedOptionsConfig.filter.enable ? (
+        mergedOptionsConfig.filter.customComponent ? (
+          mergedOptionsConfig.filter.customComponent
+        ) : filterCallback ? (
```
Comment (dv-usama-ansari) [inline src/vis/bar/BarVisSidebar.tsx:102]: Lift up the filter callback so that even the custom component wont be rendered if the filter callback is empty

Comment (dvmartinweigl) [reply src/vis/bar/BarVisSidebar.tsx:102]: I tested your latest changes @dv-usama-ansari in bioinsight, but for me the filter options are still appearing even if the callback is not implemented.

image

**Review by dvmartinweigl — APPROVED** [review]
Tested and this now works as expected. Also looks good code wise

**Pull request merged**